api/src/main/java/io/jsonwebtoken/lang/UnavailableImplementationException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2019 jsonwebtoken.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.jsonwebtoken.lang;
+
+/**
+ * Exception indicating that no implementation of an jjwt-api SPI was found on the classpath.
+ */
+public class UnavailableImplementationException extends RuntimeException {
```

Comment:
Since this is a brand new public API class - we should probably lock this down and make it `final` (open/closed principle, etc).  We can always remove the modifier later if we need to subclass it for our own needs.

Reply:
💯